src/algorithm/center-of-mass-derivatives.hxx

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2019 CNRS
+//
+
+#ifndef __pinocchio_center_of_mass_derivatives_hxx__
+#define __pinocchio_center_of_mass_derivatives_hxx__
+
+#include "pinocchio/multibody/visitor.hpp"
+#include "pinocchio/algorithm/check.hpp"
+
+namespace pinocchio
+{
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl,
+           typename Matrix3xOut>
+  struct CoMVelocityDerivativesForwardStep
+    : public fusion::JointVisitorBase< CoMVelocityDerivativesForwardStep<Scalar,Options,JointCollectionTpl,Matrix3xOut> >
+  {
+    /* Assumes that both computeForwardKinematicsDerivatives and centerOfMass (or
+     * equivalent methods like computeAllTerms) have been computed beforehand. */
+
+    typedef ModelTpl<Scalar,Options,JointCollectionTpl> Model;
+    typedef DataTpl<Scalar,Options,JointCollectionTpl> Data;
+    
+    typedef boost::fusion::vector<const Model &,
+                                  Data &,
+                                  Matrix3xOut &
+                                  > ArgsType;
+    
+    template<typename JointModel>
+    static void algo(const JointModelBase<JointModel> & jmodel,
+                     JointDataBase<typename JointModel::JointDataDerived> & jdata,
+                     const Model & model,
+                     Data & data,
+                     const Eigen::MatrixBase<Matrix3xOut> & vcom_partial_dq_CONST)
+    {
+      typedef typename Model::JointIndex JointIndex;
+      typedef typename Data::Motion Motion;
+      
+      const JointIndex & i = jmodel.id();
+
+      Matrix3xOut & dvcom_dq = PINOCCHIO_EIGEN_CONST_CAST(Matrix3xOut,vcom_partial_dq_CONST);
+      typename SizeDepType<JointModel::NV>::template ColsReturn<typename Data::Matrix3x>::Type
+        dvcom_dqi = jmodel.jointCols(dvcom_dq);
+
+      Motion vpc = (model.parents[i]>0) ? (data.v[i]-(Motion)jdata.v()) : Motion::Zero();
```

Comment:
You can put this result in data.v[i] if you want.

Reply:
thanks for the trick.